src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php

```diff
@@ -72,6 +73,12 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                         }
                     }
 
+                    if ($data['type'] === PercentageDiscountPriceCalculator::TYPE) {
+                        if ($data['configuration']['amount'] === '') {
+                            return;
+                        }
+                    }
+
```

Comment:
do we really want to hardcode these values (line 76 and 68), it looks like a workaround :(
what will happen with a new custom calculator?
I don't know any idea how to improve it :D but maybe @GSadee @Zales0123 or @lchrusciel  have some hint :)

Reply:
It should be cover with the regular validation and proper validation message